tests/baselines/reference/jsFileFunctionOverloads.symbols

```diff
@@ -0,0 +1,87 @@
+=== tests/cases/compiler/jsFileFunctionOverloads.js ===
+/**
+ * @overload
+ * @param {number} x
+ * @returns {'number'}
+ */
+/**

```

Comment:
what happens if all the `@overload` tags are in one comment? Does that work too?

Reply:
Yes, actually it does. At least the examples I checked locally seem to work properly. Do you think we should be adding this as a test case?

Reply:
Just in case, I've also added another variant of tests, which tries to put all `@overload` tags in a single comment. It seems like things get tricky (unsurprisingly) if you also want to use `@template` tag, as you can only have a single "scope" for template parameters per JSDoc comment. This problem is not specific to `@overload` tag though, and I think it should be good enough for now if it's covered in documentation properly. Perhaps this could be improved in the future if we allow `@template` to be parsed as part of `JSDocSignature`?

Reply:
Yes, I think it's good enough for now.